### ISSUE TITLE
.eslintignore shouldn't show warnings on directory, only on file.

### ISSUE THREAD

**Issue by giantelk** [eslint/eslint#6349]
**What version of ESLint are you using?**
`6.0.4`

**What parser (default, Babel-ESLint, etc.) are you using?**
`package.json`
```
  "dependencies": {
    "babel-preset-react": "6.5.0",
    "babelify": "7.3.0",
    "react": "15.1.0",
    "react-dom": "15.1.0",
    "react-router": "2.4.1"
  },
  "devDependencies": {
    "babel-eslint": "6.0.4",
    "babel-jest": "12.1.0",
    "babel-polyfill": "6.9.0",
    "babel-preset-es2015": "*",
    "eslint": "2.11.0",
    "eslint-plugin-react": "5.1.1",
    "jest-cli": "12.1.1",
    "react-addons-test-utils": "15.1.0"
  }
```

**Please show your full configuration:**

```
# .eslintignore

# /node_modules/* and /bower_components/* ignored by default

# Ignore built files.
build/*

# Ignore tests
__tests__

non-production/*
app/junk
```

**What did you do? Please include the actual source code causing the issue.**
From CLI
`node ./node_modules/.bin/eslint ./app/*/*.js`

**What did you expect to happen?**
As per docs, in .eslintignore, listing only a directory should not show warnings in console. Only show warnings if a file is ignored.
http://eslint.org/docs/user-guide/configuring#ignored-file-warnings

**What actually happened? Please include the actual, raw output from ESLint.**

> /Documents/dev/reactjs/FlyTweet/app/junk/CheckboxWithLabel.js
>   0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

**Comment by platinumazure** [comment]
Hi @giantelk, thanks for submitting this issue.

One of the subtle things about running ESLint from a command shell is that the shell may intervene and interpret your input before it even gets to ESLint. So what happens sometimes is, when a glob pattern is passed, the shell will actually expand that into a list of files; as a result, ESLint will warn about ignored files because, from its perspective, you have passed in a list of files.

This can be fixed by quoting your glob pattern with double quotes, which will prevent the shell from interpreting the glob and which will allow the actual glob string to get to ESLint.

Can you try running this command?

```shell
 node ./node_modules/.bin/eslint "./app/*/*.js"
```

**Comment by giantelk** [comment]
Problem solved, no more warnings with double quotes around `glob pattern`:
`node ./node_modules/.bin/eslint "./app/*/*.js"`

**Comment by ilyavolodin** [comment]
Closing as this is an issue with shell, not ESLint. Not much we can do about it.

**Comment by giantelk** [comment]
I suggest add a note about this to the documentation.
